src/napari/_qt/layer_controls/widgets/qt_face_color.py

```diff
@@ -42,33 +42,34 @@ def __init__(
         tooltip: Optional[str] = None,
     ) -> None:
         super().__init__(parent, layer)
-        # Setup layer
-        self._layer.events.current_face_color.connect(
-            self._on_current_face_color_change
-        )
-        if hasattr(self._layer, '_face'):
-            # Handle Points layer case
-            self._layer._face.events.current_color.connect(
-                self._on_current_face_color_change
-            )
-
         # Setup widgets
         self.face_color_edit = QColorSwatchEdit(
             initial_color=self._layer.current_face_color,
             tooltip=tooltip,
         )
         self.face_color_label = QtWrappedLabel(trans._('face color:'))
-        self._on_current_face_color_change()
         connect_setattr(
             self.face_color_edit.color_changed,
             self._layer,
             'current_face_color',
         )
-
-    def _on_current_face_color_change(self) -> None:
-        """Receive layer model face color change event and update color swatch."""
-        with qt_signals_blocked(self.face_color_edit):
-            self.face_color_edit.setColor(self._layer.current_face_color)
+        self._callbacks.append(
+            attr_to_settr(
+                self._layer,
+                'current_face_color',
+                self.face_color_edit,
+                'setColor',
+            )
+        )
+        if hasattr(self._layer, '_face'):
```

Comment:
```suggestion
        if hasattr(self._layer, '_face'):
            # Handle Points layer case
```

Copy explanation from previous code.